style_guide.md

```diff
@@ -0,0 +1,118 @@
+These are some quick guidelines and principles that this project follows. 
+
+In general...
+* **Structure** should follow the [PEP 8](https://www.python.org/dev/peps/pep-0008/#introduction) format.
+* **Docstrings** should follow the [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) format. (Sec. 3)
+* Code and comments should be kept **beginner friendly**. Purpose and diction should be easy to understand. (Sec. 4)
+* When in doubt look around; you should ***STAY CONSISTENT*** with the rest of the project. 
+
+In order to keep things beginner friendly... 
+* All classes, methods, and functions should include a docstring. (Sec. 3)
+* Loop and control structures should be documented at each line. (Sec. 4)
+
+## 2. Names
+
+### Best Naming Practices
+Names should be **descriptive**, and **concise**.
+*  **Descriptive**: Names should accurately describe the entity they represent. 
+*  **Concise**: Names should be as short as possible without extraneous abbreviation, or loss of meaning. 
+
+"Function" in this section refers to a functions, methods, and generators as group.
+
+### Names of Structures and Data
+Names should follow [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Naming#Naming) convention. Simply put...
+* **Classes** should be upper camel case, e.x. ```class MySuperCoolClass():```
+* **Functions**: should be lowercase underscored, e.x. ```def my_function():``` 
+* **Variables**: should be lowercase underscored, e.x. ```my_integer```
+
+## 3. Comments
+Some of our guidelines are contradictory to [PEP 8](https://www.python.org/dev/peps/pep-0008/#introduction) and [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) standards.
+We deviate from these formats in an effort to remain **beginner friendly**.
+
+### Inline Comments
+You should document all loop and control structures with inline comments. 
+These comments should be indented at least 8 spaces after the end of the line. 
+Indentation of the comment should match indentation of the line it describes. 
+There should be a space after each '#', and the first letter of the comment should be captialized. 
+Ex. 
+```python
+while self.isAlive is True:                # While person is alive
+    if self.age < 6 or self.isRetired:         # If too old or young
+        stay_home()                                # Stay home
+    elif self.age < 18:                        # Else if school age
+        goto_school()                              # Goto school
+    else:                                      # Otherwise
+        goto_work()                                # Goto work
+```
+
+### Block Comments 
+Block comments should be used to describe logical groupings of statements, if appropriate, but not loop or control structures. 
+They should be placed above the code block they describe, with a new line above itself and below the code block it describes. 
+Ex. 
+```python
```

Comment:
again, if we can find an example from the code, that would be really great

Reply:
We also comment slightly different than the example you have given, since we comment both on the condition and what happens - this make the comments in to full sentences, that are easier to actually understand. You can check the examples I have in the "original style guide", they are examples from the code.

Reply:
Similarly to above I used an example I felt followed the guidelines better than code currently in the project. I also updated the example to show condition and response comments.

Reply:
Here is an example from the codebase of what I mean when I say that we give full sentences that comment both on the condition and on what happens (i.e. only one line of comments for the control statement and the conditional)

for i in range(self.days):        # Color a rectangle pr. completed day blue (from left to right)
            self.canvas.itemconfig(self.rectangle_list[i], fill = "blue")

Often comments in code (and especially python code which is so readable) don't add any extra value. That is why I focus on the comments being actual "human" language of full sentences, as I believe it adds extra value.